src/context/user_context.js

```diff
@@ -0,0 +1,114 @@
+import React, { useContext, useEffect, useState } from 'react';
+import axios, { AxiosError } from 'axios';
+
+// Since this rootURL is used throughout the app, it might make sense to instead have a file for "Constants" that can be imported wherever needed
+const rootUrl = 'https://ecommerce-6kwa.onrender.com';
+
+const UserContext = React.createContext();
+
+export const UserProvider = ({ children }) => {
+  const [currentUser, setCurrentUser] = useState(null);
+  const [authLoading, setAuthLoading] = useState(false);
+  const [authError, setAuthError] = useState(false);
+
+  useEffect(() => {
+    async function fetchData() {
+      const url = `${rootUrl}/api/v1/users/showMe`;
+      axios
+        .get(url, { withCredentials: true })
+        .then((response) => {
+          console.log(response);
+          setCurrentUser(response.data.user);
+        })
+        .catch((error) => {
+          const errorPayload =
+            error instanceof AxiosError ? error.response.data : error;
+          console.error(errorPayload);
+          setAuthError(errorPayload);
+        });
+    }
+    fetchData();
+    // By using empty array [], for the "dependencies" argument of useEffect, it tells React to run this useEffect hook only *once*, the first time this component/context is rendered
+  }, []);
```

Comment:
This `useEffect` hook will run only once, because we gave it an empty dependency array. It will run whenever the `<UserContext.Provider>` component is first rendered